### PR TITLE
timerdevice_boot: add reboot test case for rtc_base="2020-06-17"

### PR DESCRIPTION
**Pull request by sitoliu**
ID: 1468140, 1468139
Signed-off-by: Sitong Liu <siliu@redhat.com>

### CODE REVIEW

File: qemu/tests/timerdevice_boot.py [hunk 1]
```diff
@@ -75,19 +76,39 @@ def verify_timedrift(session, is_hardware=False):
...
         if is_hardware:
             guest_time = get_hwtime(session)
-        drift = abs(float(host_time) - float(guest_time))
-        if drift > timerdevice_drift_threshold:
-            test.fail("The guest's %s time is different with"
-                      " host's system time. Host time: '%s', guest time:"
-                      " '%s'" % (time_type, host_time, guest_time))
+        if not base_time:
+            drift = abs(float(host_time) - float(guest_time))
+            if drift > timerdevice_drift_threshold:
```
Comment (xiangchunfu) [inline qemu/tests/timerdevice_boot.py]: @sitoliu  If base_time is "2006-06-17", then guest_time should be "2006-06-17",  drift value is much larger than 3 mins.

**Comment by xiangchunfu** [comment]
Host_RHEL.mX.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.X.0.x86_64.io-github-autotest-qemu.timerdevice.boot_test.with_reboot.reboot_immediately.without_host_load.clksource_kvm-clock.base_date.clock_host: ERROR: list index out of range (61.29 s)

product bug block this test.

**Comment by sitoliu** [comment]
> Host_RHEL.mX.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.X.0.x86_64.io-github-autotest-qemu.timerdevice.boot_test.with_reboot.reboot_immediately.without_host_load.clksource_kvm-clock.base_date.clock_host: ERROR: list index out of range (61.29 s)
> 
> product bug block this test.

Correct test fail log.

FAIL 1-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.timerdevice.boot_test.with_reboot.reboot_immediately.without_host_load.clksource_kvm-clock.base_date.clock_host -> TestFail: The guest system time is not from base time 2006-06-17T08:00:00

File: qemu/tests/timerdevice_boot.py [hunk 2]
```diff
@@ -75,19 +76,39 @@ def verify_timedrift(session, is_hardware=False):
...
+                    base_time += "T08:00:00"
+                try:
+                    time_struct = time.strptime(base_time, base_pattern)
+                except:
+                    raise
+            target_time = time.mktime(time_struct)
+            # assume boot time will not large than 3 minutes
+            if (guest_time - target_time) > 180:
+                test.fail("The guest system time is not from base time "
+                          "%s" % base_time)
```
[the remark this answers is not in the document]
Comment (sitoliu) [reply qemu/tests/timerdevice_boot.py]: @xiangchunfu Thanks, do you think 360s is ok?

[the remark this answers is not in the document]
Comment (sitoliu) [reply qemu/tests/timerdevice_boot.py]: @xiangchunfu Thanks, updated it. Could you help review it again? Thanks.

**Comment by sitoliu** [comment]
Depends on: https://github.com/avocado-framework/avocado-vt/pull/1820

File: qemu/tests/cfg/timerdevice.cfg [hunk 3]
```diff
@@ -58,6 +58,10 @@
                 - base_localtime:
                     only Windows
                     rtc_base = localtime
+                - base_date:
+                    Windows:
+                        only Win2000, WinXP, Win2003, WinVista, Win7, Win2008
+                    rtc_base = "2006-06-17"
```
Comment (YanhuiMa) [inline qemu/tests/cfg/timerdevice.cfg]: For windows guest, now we only test future time, such as "2020-05-30"

Comment (sitoliu) [reply qemu/tests/cfg/timerdevice.cfg]: Yes, it should be, as I talked in our weekly meeting.

File: qemu/tests/cfg/timerdevice.cfg [hunk 4]
```diff
@@ -58,6 +58,10 @@
                 - base_localtime:
                     only Windows
                     rtc_base = localtime
+                - base_date:
+                    Windows:
+                        only Win2000, WinXP, Win2003, WinVista, Win7, Win2008
```
Comment (YanhuiMa) [inline qemu/tests/cfg/timerdevice.cfg]: Could you pls explain why specify these windows guest here?  I think the boot and reboot cases are applicative for win2012r2, win2019, etc. So no need to only here.

**Pull request closed without merging**